File: fix_dashboard.py
import sys
import os

path = r'c:\Users\KSM\Desktop\horpay\horpaytest\src\app\dashboard\DashboardClient.tsx'

if not os.path.exists(path):
    print(f"File not found: {path}")
    sys.exit(1)

with open(path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

start_marker = "/* ── Rooms Tab Content (Premium Redesign) ── */"
end_marker = "/* ── Settings Tab Content (NEW - Consolidated Single Page) ── */"

start_idx = -1
end_idx = -1

for i, line in enumerate(lines):
    if start_marker in line:
        start_idx = i
    if end_marker in line:
        end_idx = i
        break

if start_idx != -1 and end_idx != -1:
    print(f"Found markers at {start_idx + 1} and {end_idx + 1}")
    
    # We replace from start_marker until just before end_marker.
    # The start_marker line itself should be replaced.
    
    new_content = [
        "                {/* ── Rooms Tab Content (Modular) ── */}\n",
        "                {activeTab === 'rooms' && (\n",
        "                    <RoomsTab\n",
        "                        rooms={rooms}\n",
        "                        selectedFloor={selectedFloor}\n",
        "                        selectedStatus={selectedStatus}\n",
        "                        setSelectedFloor={setSelectedFloor}\n",
        "                        setSelectedStatus={setSelectedStatus}\n",
        "                        waitingVerifyRoomIds={waitingVerifyRoomIds}\n",
        "                        unpaidRoomIds={unpaidRoomIds}\n",
        "                        overdueRoomIds={overdueRoomIds}\n",
        "                        movingOutRoomIds={movingOutRoomIds}\n",
        "                        router={router}\n",
        "                    />\n",
        "                )}\n",
        "\n"
    ]
    
    lines[start_idx:end_idx] = new_content
    
    with open(path, 'w', encoding='utf-8') as f:
        f.writelines(lines)
    print("Successfully updated DashboardClient.tsx")
else:
    print(f"Markers not found. Start: {start_idx}, End: {end_idx}")
    sys.exit(1)

File: fix_dashboard.js
const fs = require('fs');
const path = require('path');

const filePath = 'src/app/dashboard/DashboardClient.tsx';
if (!fs.existsSync(filePath)) {
    console.error(`File not found: ${filePath}`);
    process.exit(1);
}

const content = fs.readFileSync(filePath, 'utf8');
const lines = content.split('\n');

// Relaxed markers to avoid indentation issues
const startMarker = "Rooms Tab Content (Premium Redesign)";
const endMarker = "Settings Tab Content (NEW - Consolidated Single Page)";

let startIdx = -1;
let endIdx = -1;

for (let i = 0; i < lines.length; i++) {
    if (lines[i].includes(startMarker)) startIdx = i;
    if (lines[i].includes(endMarker)) {
        endIdx = i;
        break;
    }
}

if (startIdx !== -1 && endIdx !== -1) {
    console.log(`Found markers at lines ${startIdx + 1} and ${endIdx + 1}`);
    
    const newContent = [
        "                {/* ── Rooms Tab Content (Modular) ── */}",
        "                {activeTab === 'rooms' && (",
        "                    <RoomsTab",
        "                        rooms={rooms}",
        "                        selectedFloor={selectedFloor}",
        "                        selectedStatus={selectedStatus}",
        "                        setSelectedFloor={setSelectedFloor}",
        "                        setSelectedStatus={setSelectedStatus}",
        "                        waitingVerifyRoomIds={waitingVerifyRoomIds}",
        "                        unpaidRoomIds={unpaidRoomIds}",
        "                        overdueRoomIds={overdueRoomIds}",
        "                        movingOutRoomIds={movingOutRoomIds}",
        "                        router={router}",
        "                    />",
        "                )}",
        ""
    ];
    
    // Replace from startIdx until endIdx (exclusive of endIdx)
    lines.splice(startIdx, endIdx - startIdx, ...newContent);
    
    fs.writeFileSync(filePath, lines.join('\n'));
    console.log("Successfully updated DashboardClient.tsx");
} else {
    console.error(`Markers not found. Start: ${startIdx}, End: ${endIdx}`);
    process.exit(1);
}
